factcast-core/src/main/java/org/factcast/core/util/FactCastJson.java

```diff
@@ -158,10 +177,21 @@ public static JsonNode toJsonNode(Map<String, Object> jsonAsMap) {
     @SneakyThrows
     public static <A> A readValueFromBytes(Class<A> type, byte[] bytes) {
         return objectMapper.readerFor(type).readValue(bytes);
-
     }
 
     public static ObjectMapper mapper() {
         return objectMapper;
     }
+
+    @SneakyThrows
+    public static long calculateHash(Class<?> projectionClass) {
+
+        JsonSchema jsonSchema = schemaGen.generateSchema(projectionClass);
```

Comment:
also i guess, the schema should be cached (Map<Type,Schema>)

Reply:
or better Ma<Type,Long> directly to the md5

Reply:
Done in AbstractSnapshotRepository